src/rest/zone_client.erl

```diff
@@ -60,118 +61,185 @@ index_ready(Zone) when is_binary(Zone) ->
 index_ready(AllZones) when is_list(AllZones) ->
     {error, not_implemented}.
 
+% MUST
 -spec is_adjacent(Zone:: binary(), From:: binary(), To:: binary()) -> {ok, boolean()} | {error, any()}.
-is_adjacent(_Zone, _From, _To) ->
-    {error, not_implemented}.
+is_adjacent(Zone, From, To) ->
+    {ok, Address} = http_utils:get_address(Zone),
+    Url = http_utils:build_url(Address, <<"is_adjacent">>, [{<<"from">>, From}, {<<"to">>, To}]),
+    http_executor:post_request(Url, true).
 
+% SHOULD POTENTIAL
 -spec list_adjacent(Zone:: binary(), Of:: binary()) -> {ok, list(binary())} | {error, any()}.
-list_adjacent(_Zone, _Of) ->
-    {error, not_implemented}.
+list_adjacent(Zone, Of) ->
+    {ok, Address} = http_utils:get_address(Zone),
+    Url = http_utils:build_url(Address, <<"list_adjacent">>, [{<<"of">>, Of}]),
+    http_executor:post_request(Url, true).
 
+% SHOULD POTENTIAL
 -spec list_adjacent_reversed(Zone:: binary(), Of:: binary()) -> {ok, list(binary())} | {error, any()}.
-list_adjacent_reversed(_Zone, _Of) ->
-    {error, not_implemented}.
+list_adjacent_reversed(Zone, Of) ->
+    {ok, Address} = http_utils:get_address(Zone),
+    Url = http_utils:build_url(Address, <<"list_adjacent_reversed">>, [{<<"of">>, Of}]),
+    http_executor:post_request(Url, true).
 
+% MUST POTENTIAL
 -spec permissions(Zone::binary(), From:: binary(), To:: binary()) -> {ok, binary()} | {error, any()}.
-permissions(_Zone, _From, _To) ->
-    {error, not_implemented}.
+permissions(Zone, From, To) ->
+    {ok, Address} = http_utils:get_address(Zone),
+    Url = http_utils:build_url(Address, <<"permissions">>, [{<<"from">>, From}, {<<"to">>, To}]),
+    http_executor:post_request(Url, true).
 
+% MUST
 -spec add_edge(Zone:: binary(), From:: binary(), To:: binary(), Permissions:: binary(), Trace:: binary()
     ) -> ok | {error, any()}.
-add_edge(_Zone, _From, _To, _Permissions, _Trace) ->
-    {error, not_implemented}.
-
+add_edge(Zone, From, To, Permissions, Trace) ->
+    {ok, Address} = http_utils:get_address(Zone),
+    Params = [{<<"from">>, From}, {<<"to">>, To}, {<<"permissions">>, Permissions}, {<<"successive">>, <<"false">>}]
+    ++ (case Trace of undefined -> []; _ -> [{<<"trace">>, Trace}] end),
+    Url = http_utils:build_url(Address, <<"graph/edges">>, Params),
+    http_executor:post_request(Url, false).
```

Comment:
Zbędne powtórzenie kodu, lepiej wywołać `add_edge(Zone, From, To, Permissions, Trace, false)`